wrangler-service/src/main/java/co/cask/wrangler/dataset/workspace/WorkspaceDataset.java

```diff
@@ -204,6 +235,39 @@ public void deleteWorkspace(String id) throws WorkspaceException {
     }
   }
 
+  /**
+   * Deletes a group of workspaces.
+   *
+   * @param group to be deleted.
+   * @return number of workspaces deleted within a group.
+   * @throws WorkspaceException thrown if there are issues with deleting workspace within group.
+   */
+  @WriteOnly
+  public int deleteGroup(String group) throws WorkspaceException {
+    int count = 0;
+    co.cask.cdap.api.dataset.table.Row row;
+    try (Scanner scanner = table.scan(null, null)) {
```

Comment:
How big can the table get? Since it is doing a full table scan, this might never succeed on a slow cluster with a lot of entries since it executes in one transaction. Should batch deletes in separate transaction.